h/activity/query.py

```diff
@@ -123,11 +123,12 @@ def execute(request, query, page_size):
     # Add group information to buckets and present annotations
     for timeframe in result.timeframes:
         for bucket in timeframe.document_buckets.values():
+            bucket.presented_annotations = []
             for index, annotation in enumerate(bucket.annotations):
-                bucket.annotations[index] = {
+                bucket.presented_annotations.append({
```

Comment:
What this originally used to do was change `bucket.annotations` from being a `List[model.Annotation]` to `List[Dict]`. That was confusing both when reading the code and in trying to reason about what could be done with the `annotations` property in `DocumentBucket`. Here I've changed the code to build up the `List[Dict]` in a separate property.